pandas-stubs/core/indexes/datetimes.pyi

```diff
@@ -39,7 +38,7 @@ from pandas.tseries.offsets import BaseOffset
 class DatetimeIndex(DatetimeTimedeltaMixin[Timestamp], DatetimeIndexProperties):
     def __init__(
         self,
-        data: ArrayLike | AnyArrayLike | list | tuple,
+        data: AnyArrayLike | list | tuple,
```

Comment:
also drive-by: `AnyArrayLike` is more generic than `ArrayLike`, so `ArrayLike` can be removed from here, right?

Reply:
No.   From `_typing.pyi`  :
```python
ArrayLike: TypeAlias = ExtensionArray | np.ndarray[Any, Any]
AnyArrayLike: TypeAlias = Index[Any] | Series[Any] | np.ndarray[Any, Any]
```
So `ArrayLike` has `ExtensionArray` but not `Index` or `Series`

Although maybe we should change that in this PR in `_typing.py` so it corresponds to pandas so that `AnyArrayLike` is a union of `ArrayLike` with `Index[Any]` and `Series[Any]`

Reply:
ah thanks i was looking at the pandas `AnyArrayLike` and noticed that that one included `ArrayLike`

I'll leave this out of the current PR then 👍